addOns/allinonenotes/src/main/java/org/zaproxy/zap/extension/allinonenotes/ExtensionAllInOneNotes.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.allinonenotes;
+
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.swing.Box;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JScrollPane;
+import javax.swing.JToolBar;
+import org.jdesktop.swingx.JXTable;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.ExtensionHookView;
+import org.parosproxy.paros.extension.SessionChangedListener;
+import org.parosproxy.paros.extension.history.ExtensionHistory;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.model.HistoryReferenceEventPublisher;
+import org.parosproxy.paros.model.Session;
+import org.zaproxy.zap.ZAP;
+import org.zaproxy.zap.utils.TableExportButton;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionAllInOneNotes extends ExtensionAdaptor implements SessionChangedListener {
+
+    private ZapMenuItem menuReload;
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionAllInOneNotes";
+    protected static final String PREFIX = "allinonenotes";
+
+    /**
+     * Relative path (from add-on package) to load add-on resources.
+     *
+     * @see Class#getResource(String)
+     */
+    private static final String RESOURCES = "resources";
+
+    private static final ImageIcon ICON =
+            new ImageIcon(ExtensionAllInOneNotes.class.getResource(RESOURCES + "/notepad.png"));
+
+    private AbstractPanel statusPanel;
+    private NotesTableModel notesTableModel = null;
+    protected static JXTable notesTable = null;
```

Comment:
- This should be private, we should pass the model to `EventConsumerImpl` instead.
- None of these variables should be static.